plugins/mirror/bazel.go

```diff
@@ -84,6 +95,32 @@ func FilterArtifactsWithoutMirror(artifacts []Artifact, regexp *regexp.Regexp) (
 	return noMirror
 }
 
+func RemoveStaleDownloadURLS(artifacts []Artifact, ignoreURLSMatching *regexp.Regexp) {
+	for _, artifact := range artifacts {
+		for _, url := range artifact.URLs() {
+			if ignoreURLSMatching.MatchString(url) {
+				continue
```

Comment:
Should we limit the 404 removal only to artifacts where we find a mirror?

Reply:
I think so, as in case of dry-run the artifacts might not yet be present.